src/test/java/unit/service/BandServiceUnitTest.java

```
@@ -0,0 +1,108 @@
+
+        assertThrows(FailedToGetBandException.class, () -> bandService.getAllBands());
+    }
+
+    @Test
+    void getBandById_shouldReturnBand_whenDaoReturnsBand() throws SQLException, FailedToGetBandException, DatabaseConnectionException {
+        Mockito.when(bandDao.getBandById(1)).thenReturn(band);
+
+        assertEquals(band, bandService.getBandById(1));
+    }
+
+    @Test
+    void getBandById_shouldThrowFailedToGetBandException_whenDaoThrowsSQLException() throws SQLException, DatabaseConnectionException {
+        Mockito.when(bandDao.getBandById(1)).thenThrow(SQLException.class);
+
+        assertThrows(FailedToGetBandException.class, () -> bandService.getBandById(1));
+    }
+
+    @Test
+    void createBand_shouldReturnId_whenDaoReturnsId() throws SQLException, InvalidBandException, FailedToCreateBandException, DatabaseConnectionException {
+        Mockito.when(bandDao.createBand(bandRequest)).thenReturn(1);
+
+        assertEquals(1, bandService.createBand(bandRequest));
+    }
+
+    @Test
+    void createBand_shouldThrowSqlException_whenDaoThrowsSqlException() throws SQLException, DatabaseConnectionException {
+        Mockito.when(bandDao.createBand(bandRequest)).thenThrow(SQLException.class);
+
+        assertThrows(FailedToCreateBandException.class, () -> bandService.createBand(bandRequest));
+    }
+
+    @Test
+    void createBand_shouldThrowFailedToCreateBandException_whenDaoThrowsSQLException() throws SQLException, DatabaseConnectionException {
+        Mockito.when(bandDao.createBand(bandRequest)).thenThrow(SQLException.class);
+
+        assertThrows(FailedToCreateBandException.class, () -> bandService.createBand(bandRequest));
+    }
+
+    @Test
+    void createBand_shouldThrowInvalidBandException_whenValidatorThrowsInvalidBandException() {
+        bandRequest.setName("");
+        assertThrows(InvalidBandException.class, () -> bandService.createBand(bandRequest));
+    }
+}
```

rather than setting the name can you mock the validator and force it to throw the exception